pkg/pillar/hypervisor/qmp.go

```diff
@@ -91,7 +92,12 @@ func getQemuStatus(socket string) (string, error) {
 				Status     string `json:"status"`
 			} `json:"return"`
 		}
-		err = json.Unmarshal(raw, &result)
+		dec := json.NewDecoder(bytes.NewReader(raw))
+		dec.DisallowUnknownFields()
+		err = dec.Decode(&result)
+		if err != nil {
+			err = fmt.Errorf("%w; (JSON received: '%s')", err, raw)
+		}
 		return result.Return.Status, err
```

Comment:
And it might make sense to print the full input string (result) and/or its length if the Validate fails to make it visible what was returned (which could be an empty string). Want want to print that if the response is not valid json as well.

Reply:
we can, yes, but this is highly unlikely. check the generated qapi parser of the qemu, it goes through each member of the `StatusInfo` struct:
```
void visit_type_StatusInfo_members(Visitor *v, StatusInfo *obj, Error **errp)
{
    Error *err = NULL;
   .....

    visit_type_RunState(v, "status", &obj->status, &err);
    if (err) {
        goto out;
    }
```

and if 'RunState' enum has an invalid value - qemu asserts.
```
const char *qapi_enum_lookup(const QEnumLookup *lookup, int val)
{
    assert(val >= 0 && val < lookup->size);

    return lookup->array[val];
}
```

so either JSON is different, either status is valid.

Reply:
So, do you think there is no way we get garbage in the QMP socket?

Reply:
garbage means JSON is invalid and it parsing fails.

Reply:
I added the check on empty status, but as I said according to qemu sources this is highly unlikely.

Reply:
But in this case, we assume the QMP sender works always correctly. I'd play it safe.